Update to API ReadMes (#1596)
open-telemetry/opentelemetry-dotnet: src/OpenTelemetry.Api/README.md

Example:

```csharp

> activity?.SetTag("otel.status_code", StatusCode.Error);

Reviewer: This section of the document is about instrumenting a library/app using just ActivitySource. i.e without any dependency on OpenTelemetry.API package. So we cannot use `StatusCode` here. Instead it must be the string "Error".

(once .NET support status natively, this should be done directly using the .net api, until then this is a workaround)

Author: Updated. What about the other references to enum I changed in this document? If I should also revert that, I am not sure what changes I should be writing about.

Reviewer: This section cannot containing anything from OpenTelemetry packages. For status you can ask users to put "Error/Unset/Ok" strings.

activity?.SetTag("otel.status_description", "error status description");
```

> [`StatusCode`](./Trace/StatusCode.cs) is an enum with options for `Unset`,

Reviewer: StatusCodes can be the strings "Unset", "Ok", "Error" only, which corresponds respectively to `Unset`, `Ok`, `Error` from [StatusCode.cs](./Trace/StatusCode.cs)

Author: Yeah, your other comment makes sense to me now, I'll update this.

Example:

```csharp
activity?.SetTag("otel.status_code", "Error");
activity?.SetTag("otel.status_description", "error status description");
```

> Values for the StatusCode tag have to be the strings "Unset", "Ok", or "Error",

Reviewer: ```suggestion
Values for the StatusCode tag have to be the strings "Unset", "Ok", or "Error",
```
Values for the StatusCode tag must be one of the strings "Unset", "Ok", or "Error",

> Values for the StatusCode tag have to be the strings "Unset", "Ok", or "Error",

Reviewer: ```suggestion
Values for the StatusCode tag must be one of the strings "Unset", "Ok", or "Error",
```
